dvc/info.py

```diff
@@ -136,11 +136,14 @@ def _get_supported_remotes():
 
 
 def get_fs_type(path):
-
-    partition = {
-        pathlib.Path(part.mountpoint): (part.fstype + " on " + part.device)
-        for part in psutil.disk_partitions(all=True)
-    }
+    partition = {}
+    for part in psutil.disk_partitions(all=True):
+        if part.fstype != "":
+            try:
+                mountpoint = pathlib.Path(part.mountpoint).resolve()
+                partition[mountpoint] = part.fstype + " on " + part.device
+            except PermissionError:
```

Comment:
Could you elaborate what `part.mountpoint` this happens for you? Just curious what causes this.

Reply:
Hey ! When I try to **_resolve()_** some partitions from psutil.disk_partitions, I get this error on Windows+NAS:

`ERROR: unexpected error - [WinError 21] Le p▒riph▒rique n▒est pas pr▒t: 'D:\\'
Traceback (most recent call last):
  File "c:\users\chetoua\miniconda3\lib\site-packages\dvc\main.py", line 55, in main
    ret = cmd.do_run()
  File "c:\users\chetoua\miniconda3\lib\site-packages\dvc\command\base.py", line 64, in do_run
    return self.run()
  File "c:\users\chetoua\miniconda3\lib\site-packages\dvc\command\version.py", line 13, in run
    dvc_info = get_dvc_info()
  File "c:\users\chetoua\miniconda3\lib\site-packages\dvc\info.py", line 43, in get_dvc_info
    fs_type = get_fs_type(repo.odb.local.cache_dir)
  File "c:\users\chetoua\miniconda3\lib\site-packages\dvc\info.py", line 136, in get_fs_type
    mountpoint = pathlib.Path(part.mountpoint).resolve()
  File "c:\users\chetoua\miniconda3\lib\pathlib.py", line 1204, in resolve
    s = self._flavour.resolve(self, strict=strict)
  File "c:\users\chetoua\miniconda3\lib\pathlib.py", line 205, in resolve
    s = self._ext_to_normal(_getfinalpathname(s))
PermissionError: [WinError 21] Le p▒riph▒rique n▒est pas pr▒t: 'D:\\'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\users\chetoua\miniconda3\lib\runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "c:\users\chetoua\miniconda3\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\CHETOUA\Miniconda3\Scripts\dvc.exe\__main__.py", line 7, in <module>
  File "c:\users\chetoua\miniconda3\lib\site-packages\dvc\main.py", line 88, in main
    dvc_info = get_dvc_info()
  File "c:\users\chetoua\miniconda3\lib\site-packages\dvc\info.py", line 43, in get_dvc_info
    fs_type = get_fs_type(repo.odb.local.cache_dir)
  File "c:\users\chetoua\miniconda3\lib\site-packages\dvc\info.py", line 136, in get_fs_type
    mountpoint = pathlib.Path(part.mountpoint).resolve()
  File "c:\users\chetoua\miniconda3\lib\pathlib.py", line 1204, in resolve
    s = self._flavour.resolve(self, strict=strict)
  File "c:\users\chetoua\miniconda3\lib\pathlib.py", line 205, in resolve
    s = self._ext_to_normal(_getfinalpathname(s))
PermissionError: [WinError 21] Le p▒riph▒rique n▒est pas pr▒t: 'D:\\'
(base)
`
That's why I added the try: except: block on the PermissionError, so that I don't take into account the partitions that I don't have access to among the psutil.disk_partitions

Reply:
Interesting. But makes sense so far :+1: